pkgs/development/tools/godot/mono/make-deps.nix

```diff
@@ -0,0 +1,71 @@
+{ godot-mono
+, lib
+, nuget-to-nix
+, writeScriptBin
+, path
+, stdenv
+}:
+
+# This can be used to recreate deps.nix. This entails doing things that are not permitted
+# with a sandboxed Nix build, so instead of building the deps.nix directly, it builds a
+# script that can be run to generate it.  Thus, to use this derivation, simply do:
+#
+# $ nix run <path-to-nixpkgs>#godot-mono.make-deps [outdir]
+#
+# The work will all be done in the specified outdir, and the resulting deps.nix will be
+# placed in it at the end. If you don't provide an outdir, a temporary one will be created
+# in your tmpfs. The final line of output will specify the path to the resulting deps.nix.
```

Comment:
When I was testing, I ran out of space in my nix-shell tmpfs until I deleted the folders from past runs.  I figured having the _option_ to run in a specified directory would provide a workaround in case someone's tmpfs is smaller or something.

Reply:
The issue behind that is that you are not cleaning up after a script run that may have failed. We commonly solve this with a TRAP after creating the tmp directory that removes it.